website/server/libs/cron.js

```diff
@@ -237,8 +237,43 @@ export function cron (options = {}) {
     }
   });
 
-  // move singleton Habits towards yellow.
-  tasksByType.habits.forEach((task) => { // slowly reset 'onlies' value to 0
+  // check if we've passed a day on which we should reset the habit counters, including today
+  let resetWeekly = false;
+  let resetMonthly = false;
+  for (let i = 0; i <= daysMissed; i++) {
+    if (resetWeekly === true && resetMonthly === true) {
+      break;
+    }
+    let thatDay = moment(now).subtract({days: i}).toDate();
+    if (thatDay.getDay() === 1) {
+      resetWeekly = true;
+    }
+    if (thatDay.getDate() === 1) {
+      resetMonthly = true;
+    }
+  }
+
+  tasksByType.habits.forEach((task) => {
+    // reset counters if appropriate
+
+    // this enormously clunky thing brought to you by lint
+    let reset = false;
+    if (task.frequency === 'daily') {
+      reset = true;
+    }
+    if (task.frequency === 'weekly' && resetWeekly === true) {
+      reset = true;
+    }
+    if (task.frequency === 'monthly' && resetMonthly === true) {
+      reset = true;
+    }
+    if (reset === true) {
+      task.counterUp = 0;
+      task.counterDown = 0;
+    }
+
+    // slowly reset 'onlies' value to 0
```

Comment:
I'm sorry, I feel like I need to reiterate: this is not my comment. I did not write it and I did not write the code that the comment is describing. I just MOVED the comment onto a new line because it refers to code that is now further down in the file. That is the only reason it is showing as modified in the pullreq. It has nothing to do with my code or this issue, and it presumably passed review before now since it was already in the codebase.

If you want it changed, I am happy as a convenience to you guys to change it as long as I'm in the file anyway, but in that case, please can one of you just tell me what you want it to say? Otherwise I would appreciate if review of my pullreq didn't get hung up on an old comment someone else wrote. 

I add, I am completely happy to rewrite any of my comments that you find unclear. This is just not one of them.